src/sage/schemes/curves/plane_curve_arrangement.py

```diff
@@ -0,0 +1,1246 @@
+
+        OUTPUT:
+
+        The base ring of the curve arrangement.
+
+        EXAMPLES::
+
+            sage: L.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: L.base_ring()
+            Rational Field
+        """
+        return self._base_ring
+
+    def coordinate_ring(self):
+        """
+        Return the coordinate ring.
+
+        OUTPUT:
+
+        The coordinate ring of the curve arrangement.
+
+        EXAMPLES::
+
+            sage: L.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: L.coordinate_ring()
+            Multivariate Polynomial Ring in x, y over Rational Field
+        """
+        return PolynomialRing(self._base_ring, self._names)
+
+    def change_ring(self, base_ring):
+        """
+        Return curve arrangements over a different base ring.
+
+        INPUT:
+
+        - ``base_ring`` -- a ring; the new base ring.
+
+        OUTPUT:
+
+        A new :class:`AffinePlaneCurveArrangements` instance over the new
+        base ring.
+
+        EXAMPLES::
+
+            sage: L.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: L.gen(0)
+            x
+            sage: L.change_ring(RR).base_ring()
+            Real Field with 53 bits of precision
+
+        TESTS::
+
+            sage: L.change_ring(QQ) is L
+            True
+        """
+        return AffinePlaneCurveArrangements(base_ring, names=self._names)
+
+    @cached_method
+    def ambient_space(self):
+        """
+        Return the ambient space.
+
+        EXAMPLES::
+
+            sage: L.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: L.ambient_space()
+            Affine Space of dimension 2 over Rational Field
+        """
+        return AffineSpace(self.base_ring(), 2, self._names)
+
+    def _repr_(self):
+        """
+        Return a string representation.
+
+        OUTPUT:
+
+        A string.
+
+        EXAMPLES::
+
+            sage: L.<x, y> = AffinePlaneCurveArrangements(QQ);  L
+            Curve arrangements in Affine Space of dimension 2 over Rational Field
+        """
+        return 'Curve arrangements in {0}'.format(self.ambient_space())
+
+    def _element_constructor_(self, *args, **kwds):
+        """
+        Construct an element of ``self``.
+
+        INPUT:
+
+        - ``*args`` -- positional arguments, each defining a curve
+
+        EXAMPLES::
+
+            sage: L.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: A = L._element_constructor_(x, y); A
+            Arrangement (x, y) in Affine Space of dimension 2 over Rational Field
+            sage: L._element_constructor_([x, y]) == A
+            True
+            sage: L._element_constructor_(Curve(x), Curve(y)) == A
+            True
+            sage: L._element_constructor_(y, x) == A
+            False
+       """
+        if len(args) == 1 and not (isinstance(args[0], (tuple, list))):
+            arg = (args[0], )
+        elif len(args) == 1:
+            arg = tuple(args[0])
```

Comment:
I had issues with your suggestion

Reply:
I am surprised by this. It should be possible to just do the `len(args) == 1` check once and this is just a regrouping into more logical parts (at least to me). I don't see how this would cause a change in the behavior.

Reply:
I erased an extra else, now it works.